### PR TITLE
Use `/proc/self/mountinfo` as a fallback for `docker inspect` if using `HOSTNAME` fails

### LINKED ISSUE #1321: Act: Build fails with `sh: 1: cargo: not found`
Hello, I've just finished the new version of my project. But when I did the PR an error occurred in the workflow. So I decided to revert the merge and tested the workflow locally with Act. Unfortunately, I discovered that, locally, I had one more error.
For Information MacOS isn't running with Act.

The cross version is the `v0.2.5`

The Error
```
| [command]cross build --release --target x86_64-pc-windows-gnu
| info: downloading component 'rust-src'
| info: installing component 'rust-src'
| sh: 1: cargo: not found
```
The Job:
```yml
deploy_building:
  name: Deploy Building
  needs: [ test ]
  strategy:
    matrix:
      os: [ ubuntu-latest ] # Default os for build
      target: [ aarch64-unknown-linux-gnu,
                i686-pc-windows-gnu, x86_64-pc-windows-gnu,
                i686-unknown-linux-gnu, x86_64-unknown-linux-gnu,
                x86_64-apple-darwin ]
      exclude: # Do not build for macOS on Linux
        - target: x86_64-apple-darwin
          os: ubuntu-latest
      include:
        - target: x86_64-apple-darwin
          os: macos-latest
  # Runs on latest ubuntu by default except for windows targets
  runs-on: ${{ matrix.os }}
  steps:
    - name: Checkout
      uses: actions/checkout@v1
    - name: Install rust
      uses: actions-rs/toolchain@v1
      with:
        toolchain: stable
        profile: minimal
        override: true
        target: ${{ matrix.target }}
    - name: Build release
      uses: actions-rs/cargo@v1
      with:
        use-cross: true
        command: build
        args: --release --target ${{ matrix.target }}
```

[Here ](https://github.com/cross-rs/cross/files/12454570/act_log_verbose_one_target.txt)  is the verbose log for only one target (windows x86_64)

Thanks to all the people who will try to solve my problem

### PR DESCRIPTION
**Pull request by thirteenowls**
Fixes #1321.

### CODE REVIEW

**Comment by Emilgardis** [comment]
this is awesome! Isn't the description a bit misleading though? The fallback is actually docker now, not proc?

**Comment by thirteenowls** [comment]
Oh, you're right, it is ambiguous. Does "Use `/proc/self/mountinfo` as a fallback if `docker inspect $HOSTNAME` fails" seem better to you?
(just in case, proc *is* the fallback, it first tries to use the hostname as usual and only reads mountinfo if that doesn't work)

**Comment by Emilgardis** [comment]
I think that works!

File: src/docker/shared.rs [hunk 1]
```diff
@@ -1351,19 +1351,75 @@ pub fn get_image(
...
+        if out.status.success() {
+            Ok(out.stdout()?)
+        } else {
+            let val = serde_json::from_slice::<serde_json::Value>(&out.stdout);
+            if let Ok(val) = val {
+                if let Some(array) = val.as_array() {
+                    // `docker inspect` completed but returned an empty array, most
+                    // likely indicating that the hostname isn't a valid container ID.
+                    if array.is_empty() {
+                        return docker_inspect_self_mountinfo(engine, msg_info);
```
Comment (Emilgardis) [inline src/docker/shared.rs:1401]: can you add a msg_info.debug here similar to L1413

**Review by Emilgardis — CHANGES_REQUESTED** [review]

**Comment by thirteenowls** [comment]
Done!

**Review by Emilgardis — APPROVED** [review]

**Pull request merged**